### PR TITLE
Fix/7_tutorial_skippable

### PR DESCRIPTION
**Pull request by raulgonzalezupc**
What's new?
- Now you can skip the tutorial and conversations of the ally on level 1
- Fix bug that you can cast the shield on a conversation
- When teleported you get the lightball

### CODE REVIEW

**Comment by yalania** [comment]
- I think the control are the other side, it said press 'Z' for skipping but you have to press 'X' 
- We need final art for this. and probably some buttons with a cursor instead of a press Z or X
- If you skip the tutorial the ally is not teleported to the second  zone so the placeholder mesh is shown, you may can remove the mesh component or disable it.
- I can't pass from first zone.
- You already got the lightball when teleported what is need on that part?

File: Scripts/src/Script/AllyTrigger.cpp [hunk 1]
```diff
@@ -188,3 +223,39 @@ void AllyTrigger::InitPublicGameObjects()
...
+		player_1->player_controller->SetIsInsideTrigger(false);
+		player_2->player_controller->SetIsInsideTrigger(false);
+	}
+	if (input_manager->GetKeyDown(KeyCode::X))
+	{
+		ally->do_tutorial = false;
+		ally->PlayAnimation(DialogueFeeling::FINISH);
+		tutorial_helper->tutorial_request->Disable();
+		ally->tutorial_request = false;
+		SetPlayersPositions(player_1_inside, player_2_inside);
```
Comment (yalania) [inline Scripts/src/Script/AllyTrigger.cpp:255]: In this case you I think you can skip the SetPlayersPositions

File: Scripts/src/Script/DialogueManager.cpp [hunk 2]
```diff
@@ -159,7 +159,10 @@ bool DialogueManager::IsOnDialogue() const
 {
 	return on_dialogue;
 }
-
+void GetSome(DialogueFeeling feeling)
+{
+	int i = 0;
```
Comment (yalania) [inline Scripts/src/Script/DialogueManager.cpp:164]: you can remove this.

**Review by yalania — CHANGES_REQUESTED** [review]
There are some needed changes to do, but good job :D

File: Scripts/src/Script/AllyInteractionsScript.h [hunk 3]
```diff
@@ -58,6 +59,17 @@ class AllyInteractionsScript : public Script
 	void TransformIntoLightBall();
 	bool ally_is_lightball = false;
 
+	//tutorial first pop up
+	void TutorialSkippablePopUp();
+	InputManager* GetCurrentInput();
+	TutorialHelper* GetTutorialHelper();
```
Comment (yalania) [inline Scripts/src/Script/AllyInteractionsScript.h:65]: use const for getter methods in general

File: Scripts/src/Script/AllyInteractionsScript.cpp [hunk 4]
```diff
@@ -504,5 +504,23 @@ void AllyInteractionsScript::CheckShieldTutorial()
...
+
+TutorialHelper* AllyInteractionsScript::GetTutorialHelper()
+{
+	return tutorial_helper.get();
+}
 
+InputManager* AllyInteractionsScript::GetCurrentInput()
+{
+	return input_manager;
+}
```
Comment (Dartemiss) [inline Scripts/src/Script/AllyInteractionsScript.cpp:525]: Const on all getters

File: Scripts/src/Script/AllyTrigger.cpp [hunk 5]
```diff
@@ -54,31 +59,61 @@ void AllyTrigger::Update()
...
+		else if(ally->do_tutorial && !ally->tutorial_request)
 		{
-			SetPlayersPositions(player_1_inside, player_2_inside);	
+			App->engine_log->Log("do tutorial");
+			
+			if (complains)
+			{
+				Complain(player_1_inside, player_2_inside);
+				//TODO RAUL -> before complains, put a bool that checks if the player has seen the tutorial pop up, if false, then show the pop up, if true, then check if the
+				// player wants to do the tutorial or not, if the player doesn't want the tutorial, put complains to false, if he wants, then everything remains the same.
```
Comment (Dartemiss) [inline Scripts/src/Script/AllyTrigger.cpp:82]: Is this TODO done?

File: Scripts/src/Script/AllyTrigger.h [hunk 6]
```diff
@@ -20,21 +25,28 @@ class AllyTrigger : public Script
 	void Load(const Config& config) override;
 
 	void InitPublicGameObjects() override;
+	inline void SetComplain(bool complain) {this->complains = complain; };
```
Comment (Dartemiss) [inline Scripts/src/Script/AllyTrigger.h:28]: put this on cpp

File: Scripts/src/Script/AllyTrigger.cpp [hunk 7]
```diff
@@ -188,3 +223,39 @@ void AllyTrigger::InitPublicGameObjects()
...
+
+void AllyTrigger::SetTutorialSkippable(bool player_1_inside, bool player_2_inside)
+{
+	player_1->player_controller->SetIsInsideTrigger(true);
+	player_2->player_controller->SetIsInsideTrigger(true);
+	tutorial_helper->tutorial_request->Enable();
+	ally->TutorialSkippablePopUp();
+	player_1->player_controller->can_move = false;
+	player_2->player_controller->can_move = false;
+	if (input_manager->GetKeyDown(KeyCode::Z))
```
Comment (Dartemiss) [inline Scripts/src/Script/AllyTrigger.cpp:235]: This is too harcoded, u should use a GameInput for being able to do it with gamecontroller

**Review by Dartemiss — CHANGES_REQUESTED** [review]
GJ! But there are several things that doesn't work.
- As anabel said, we should put final art as soon as possible and on game you don't know which buttons you have to press for skipping tutorial
- Also as I commented on code, the skip buttons should be GameInputs allowing players with controllers to skip the tutorial
- As anabel commented aswell, when you skip the tutorial the ally keeps on the tree and is quite contradictory if he becomes the lightball and also a white spheere mesh is visible on the second position of the ally. My suggestion would be disable that mesh and maybe give the lightball at the moment that you skip the tutorial and make the ally disapear aswell.
- Lastly, on this branch the game is bugged on battle zone1 and only the first wave spawns, making a game crashing bug and player cannot pass that part .

**Comment by yalania** [comment]
Now when I press the buttons nothing happens. And I already did the get lightball when teleporting, so it weird it was not working, can you point me to the place were you fixed it on the code?

**Review by yalania — APPROVED** [review]
Ok, needed to get latest develop in engine, my bad

**Comment by raulgonzalezupc** [comment]
> Now when I press the buttons nothing happens. And I already did the get lightball when teleporting, so it weird it was not working, can you point me to the place were you fixed it on the code?

tbh, i didn't know that you did that, I just thought since I gave the lightball to the player when enter the zone 1, why not give it to him when teleporting, so I did haha

**Pull request closed without merging**